projects/core/src/site-context/config/config-loader/site-context-config-loader.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from '@angular/core';
+import { filter, map, take } from 'rxjs/operators';
+import { BaseSite } from '../../../model/misc.model';
+import { BaseSiteService } from '../../facade/base-site.service';
+import {
+  BASE_SITE_CONTEXT_ID,
+  CURRENCY_CONTEXT_ID,
+  LANGUAGE_CONTEXT_ID,
+  THEME_CONTEXT_ID,
+} from '../../providers/context-ids';
+import { SiteContextConfig } from '../site-context-config';
+
+@Injectable({ providedIn: 'root' })
+export class SiteContextConfigLoaderService {
+  constructor(protected baseSiteService: BaseSiteService) {}
+
+  /**
+   * Initializes the SiteContextConfig asynchronously basing on the external config
+   */
+  loadConfig(): Promise<SiteContextConfig> {
+    return this.baseSiteService
+      .get()
+      .pipe(
+        filter((baseSite: any) => {
+          if (!baseSite) {
+            throw new Error(
+              `Error: Cannot get base site config! Current url doesn't match any of url patterns of any base sites.`
+            );
+          }
+          return Boolean(baseSite);
+        }),
+        map((baseSite) => this.toSiteContextConfig(baseSite)),
+        take(1)
+      )
+      .toPromise();
+  }
+
+  toSiteContextConfig(source: BaseSite): SiteContextConfig {
```

Comment:
1. Can we add add `protected`, please?
2. what about simply `getConfig(baseSite: BaseSite): SiteContextConfig`?